heltour/tournament/admin.py

```diff
@@ -45,19 +47,129 @@ class _BaseAdmin(VersionAdmin):
     change_form_template = 'tournament/admin/change_form_with_comments.html'
     history_latest_first = True
 
+    league_id_field = None
+    league_competitor_type = None
+    allow_all_staff = False
+
+    def has_assigned_perm(self, user, perm_type):
+        return 'tournament.%s_%s' % (perm_type, self.opts.model_name) in user.get_all_permissions()
+
+    def has_league_perm(self, user, action, obj):
+        if self.league_id_field is None:
+            return False
+        authorized_leagues = self.authorized_leagues(user)
+        if self.league_competitor_type is not None \
+                and all((League.objects.get(pk=pk).competitor_type != self.league_competitor_type for pk in authorized_leagues)):
+            return False
+        if obj is None:
+            return len(authorized_leagues) > 0
```

Comment:
This can be done with: return bool(authorized_leagues) not sure which reads better. I'm fine with either